File: dotnet/Allmark/Types/Delimiter.cs
namespace Allmark.Types;

/// <summary>
/// Represents a delimiter found during inline parsing.
/// </summary>
public record Delimiter
{
    public required string Markup { get; set; }
    public required int Start { get; set; }
    public required int Length { get; set; }
    public bool Handled { get; set; }
    /// <summary>
    /// Precedence for delimiter matching (higher = takes precedence).
    /// </summary>
    public int? Precedence { get; set; }
}

File: zig/src/types/Delimiter.zig
const std = @import("std");

pub const Delimiter = struct {
    markup: [3]u8,
    markup_len: u8,
    start: usize,
    length: usize,
    handled: bool = false,
    /// Precedence for delimiter matching (higher = takes precedence).
    precedence: ?u8 = null,

    pub fn getMarkup(self: *const Delimiter) []const u8 {
        return self.markup[0..self.markup_len];
    }
};
